packages/linkedom-crawler/src/internals/linkedom-crawler.ts

```diff
@@ -193,34 +200,29 @@ export class LinkeDOMCrawler extends HttpCrawler<LinkeDOMCrawlingContext> {
                     finalRequestUrl: crawlingContext.request.loadedUrl,
                 });
             },
-        };
-    }
+            async waitForSelector(selector: string, timeoutMs = 5_000) {
+                const $ = cheerio.load(this.body);
```

Comment:
are we sure that `this` in the context extensions will always be bound to the context?

Reply:
Well, in this particular case, we want to recursively call the `waitForSelector` method that we are just adding, so we cannot access it via `crawlingContext` (because of types). But we also want to allow overrides in later stages of the pipeline, so we don't want to reference the method directly.

In my opinion, `this` is the cleanest option here. But I'll check all other cases where we use it and change it to something less brittle where it makes sense.